Elaborate on matchAll's return value (#31187)
mdn/content: files/en-us/web/javascript/reference/global_objects/string/matchall/index.md

### Return value

> An [iterable iterator object](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) (which is not restartable) of matches or an empty iterator if no matches are found. Each iterator value has the same shape as {{jsxref("RegExp.prototype.exec()")}}'s return value.

Reviewer: ```suggestion
An [iterable iterator object](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) (which is not restartable) of matches or an empty iterator if no matches are found. Each value yielded by the iterator has the same shape as {{jsxref("RegExp.prototype.exec()")}}'s return value.
```

I read through your Discord reply and I'm sold; I think this redundancy is good in this case. Just one minor suggestion.

Author: Good idea. Definitely an improvement.

> An [iterable iterator object](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) (which is not restartable) of matches or an empty iterator if no matches are found. Each value yielded by the iterator has the same shape as {{jsxref("RegExp.prototype.exec()")}}'s return value.

Reviewer: ```suggestion
An [iterable iterator object](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator) (which is not restartable) of matches or an empty iterator if no matches are found. Each value yielded by the iterator is an array with the same shape as the return value of {{jsxref("RegExp.prototype.exec()")}}.
```
